### DISCUSSION TITLE
Using hpx::thread::sleep_for(1s) results in CPU utilization reaching 80%

### DISCUSSION THREAD

**Discussion by phil-skillwon** [STEllAR-GROUP/hpx#6470] [Q&A] [answered]
Using `hpx::thread::sleep_for(1s)` results in CPU utilization reaching 80%, more precisely, it causes each of the 48 CPU cores to reach an 80% utilization rate. How can this be resolved?

**Comment by Pansysk75** [comment, marked as answer]
HPX, on its default configuration, spawns one OS thread for each core in your machine as soon as the runtime is initialized. These worker threads are constantly checking for available work (literally on a while loop, see [here](https://github.com/STEllAR-GROUP/hpx/blob/3add53814f8e2ee2665c7e23af87c345c0841ffb/libs/core/thread_pools/include/hpx/thread_pools/scheduling_loop.hpp#L141)).
That explains the high utilization. The worker threads will eventually automatically suspend when there is no work. This can be controlled using the [hpx.max_idle_backoff_time](https://hpx-docs.stellar-group.org/latest/html/manual/starting_the_hpx_runtime.html#automatically-suspending-worker-threads) configuration value.

On the hpx::thread::sleep_for(), this will only suspend the current "HPX-thread", which is more like a user-space task. If you are interested in pausing the HPX runtime, this is described [here](https://hpx-docs.stellar-group.org/latest/html/manual/starting_the_hpx_runtime.html#suspending-and-resuming-the-hpx-runtime).

**Reply by phil-skillwon** [reply]
I really appreciate your help. Thank you!

**Reply by phil-skillwon** [reply]
@Pansysk75 
For the `max_idle_loop_count` and `max_idle_backoff_time` configuration options, setting them to 1000 can reduce CPU usage when idle but may impact data exchange functionality between nodes. Do you have recommended values for these two settings?

**Reply by Pansysk75** [reply]
@phil-skillwon Hmm that is a good question, I've never strayed from the defaults because the idle CPU usage hasn't been a concern. It sounds like something you'll have to investigate for your specific use case, unfortunately I can't be more helpful here.
